### PR TITLE
Fixes #120: control the lifecycle of the tcpListener socket on the availability of the tcpConnector

### LINKED ISSUE #120: TCP adaptor listener connections hang if no connectors are defined
If a connector is not defined but a listener is defined:

    A client connects to the listener. Listener starts with SYN, SYN/ACK, ACK.
    The client sends the first payload bytes
    No further traffic is passed on the client connection
    Even if the server is started while the client is waiting no further traffic is passed on to the client.
    The client seems to hang

If there is no address in the address table, close the client connection. Also free all the connection related resources in the skupper-router.

### PR DESCRIPTION
**Pull request by kgiusti**

### CODE REVIEW

**Comment by kgiusti** [comment]
DO NOT MERGE:

Disables some tests that fail - needs fixing before it can land.

File: src/adaptors/tcp_adaptor.c
Comment (ted-ross) [inline src/adaptors/tcp_adaptor.c:1418]: I think you could safely ignore in-process consumers here.

**Review by ted-ross — APPROVED** [review]

**Pull request merged**
